src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1370,7 +1370,10 @@ bool Creature::CanStartAttack(Unit const* who, bool force) const
     if (IsCivilian())
         return false;
 
-    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC))
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC) && who->GetTypeId() != TYPEID_PLAYER)  // flag is valid only for non player characters
+        return false;
+
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC) && who->GetTypeId() == TYPEID_PLAYER)  // immune to PC and target is a player, return false
         return false;
```

Comment:
I just tested your theory by removing the existing lines 1373 + 1374 (red and white lines above) and the Sunstrider Guardians react correctly by attacking in that compile. They continued to attack my hunter character's pet after switching to GM ON, though. I thought that a player combat pet would follow the player character's GM state.